x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -239,6 +261,49 @@ func (m *MetricSet) createCloudWatchEvents(getMetricDataResults []cloudwatch.Met
 	return events, nil
 }
 
+func calculateRate(resultMetricsetFields common.MapStr, monitoringState string) {
+	var period = 300.0
+	if monitoringState != "disabled" {
+		period = 60.0
+	}
+
+	networkIn, err := resultMetricsetFields.GetValue("network.in.bytes")
+	if err == nil && networkIn != nil {
+		networkInRate := networkIn.(float64) / period
+		resultMetricsetFields.Put("network.in.bytes_per_sec", networkInRate)
+	}
+
+	networkOut, err := resultMetricsetFields.GetValue("network.out.bytes")
+	if err == nil && networkOut != nil {
+		networkOutRate := networkOut.(float64) / period
+		resultMetricsetFields.Put("network.out.bytes_per_sec", networkOutRate)
+	}
+
+	diskReadBytes, err := resultMetricsetFields.GetValue("diskio.read.bytes")
+	if err == nil && networkOut != nil {
+		diskReadBytesRate := diskReadBytes.(float64) / period
+		resultMetricsetFields.Put("diskio.read.bytes_per_sec", diskReadBytesRate)
+	}
+
+	diskWriteBytes, err := resultMetricsetFields.GetValue("diskio.write.bytes")
+	if err == nil && networkOut != nil {
+		diskWriteBytesRate := diskWriteBytes.(float64) / period
+		resultMetricsetFields.Put("diskio.write.bytes_per_sec", diskWriteBytesRate)
+	}
+
+	diskReadOps, err := resultMetricsetFields.GetValue("diskio.read.count")
+	if err == nil && networkOut != nil {
+		diskReadOpsRate := diskReadOps.(float64) / period
+		resultMetricsetFields.Put("diskio.read.count_per_sec", diskReadOpsRate)
+	}
+
+	diskWriteOps, err := resultMetricsetFields.GetValue("diskio.write.count")
+	if err == nil && networkOut != nil {
+		diskWriteOpsRate := diskWriteOps.(float64) / period
+		resultMetricsetFields.Put("diskio.write.count_per_sec", diskWriteOpsRate)
+	}
```

Comment:
A lot of repeated code, could we put it in a loop or a function?

Reply:
yep, done 👍